Docs: Concurrency Check & `IHasConcurrencyStamp` Interface

EngincanV opened the pull request:
Resolves #8425

EngincanV committed:
Docs: Update Concurrency-Check.md

EngincanV committed:
Update docs-nav.json

EngincanV committed:
Identity: Update Concurrency Failure localization value

rqx110 commented on modules/identity/src/Volo.Abp.Identity.Domain.Shared/Volo/Abp/Identity/Localization/zh-Hans.json:
```diff
@@ -37,7 +37,7 @@
     "PersonalSettingsSavedMessage": "你的个人设置保存成功.",
     "Volo.Abp.Identity:DefaultError": "发生了一个未知错误.",
-    "Volo.Abp.Identity:ConcurrencyFailure": "对象已被修改,乐观并发失败.",
+    "Volo.Abp.Identity:ConcurrencyFailure": "乐观并发检查失败。 您正在处理的实体已被其他用户修改。 请放弃您的更改，然后重试。",
```
```suggestion
    "Volo.Abp.Identity:ConcurrencyFailure": "乐观并发检查失败. 你正在处理的对象已被其他用户修改. 请放弃你的更改, 然后重试.",
```

rqx110 commented on modules/identity/src/Volo.Abp.Identity.Domain.Shared/Volo/Abp/Identity/Localization/zh-Hant.json:
```diff
@@ -37,7 +37,7 @@
     "PersonalSettingsSavedMessage": "你的個人設置保存成功.",
     "Volo.Abp.Identity:DefaultError": "發生了一個未知錯誤.",
-    "Volo.Abp.Identity:ConcurrencyFailure": "對象已被修改,因樂觀並行控制導致失敗.",
+    "Volo.Abp.Identity:ConcurrencyFailure": "樂觀並發檢查失敗。 您正在處理的實體已被其他用戶修改。 請放棄您的更改，然後重試。",
```
```suggestion
    "Volo.Abp.Identity:ConcurrencyFailure": "樂觀並行控制檢查失敗. 你正在處理的對象已被其他用戶修改. 請放棄你的更改, 然後重試.",
```

EngincanV committed:
Identity: Update Chinese localizations for ConcurrencyFailure

EngincanV commented:
Thanks @rqx110, I've updated the Chinese localizations.

rqx110 commented:
Actually, I don't think the end user know what is **Optimistic concurrency check** mean in the exception message.
I suggest change the exception message to "The data you submitted has been modified by another user. Please discard your changes and try again."

braim23 committed:
Update Concurrency-Check.md

in line 45 I changed the semicolon to a colon, if it's wrong you can revert it bro
